### ISSUE TITLE
Embedding a right-aligned JProgressBar in title pane blocks the title

### ISSUE THREAD

**Issue by grimlock81** [JFormDesigner/FlatLaf#272]
First of all, I love the changes in 1.1 regarding the Windows 10 native support for the frames. I have been trying to integrate them into my proejct.

As part of https://github.com/JFormDesigner/FlatLaf/pull/268 it was suggested that to add and right-align a custom component in a embedded menu bar, add `Box.createHorizontalGlue()` before adding the component. The example used was a JButton and this works perfectly okay.

However, when I try to add a JProgressBar using this same trick, I find that the progress bar ends up covering part of the Window title. I think there's something I need to do to the width of the progress bar, but setting the preferred size doesn't seem to change it. 

![](https://i.imgur.com/q6swHFq.png)

This is my test case

```Java
import com.formdev.flatlaf.intellijthemes.FlatArcIJTheme;
import java.awt.EventQueue;
import java.awt.Frame;
import javax.swing.Box;
import javax.swing.JFrame;
import javax.swing.JMenu;
import javax.swing.JMenuBar;
import javax.swing.JProgressBar;
import javax.swing.UIManager;
import javax.swing.UnsupportedLookAndFeelException;

public class TestEmbeddingProgressBar {
    public static void main(String[] args) throws Exception {
        EventQueue.invokeLater(() -> {
            startGUI();
        });
    }

    private static void startGUI() {
        try {
            UIManager.setLookAndFeel(new FlatArcIJTheme());
        }
        catch (UnsupportedLookAndFeelException ex) {
            ex.printStackTrace();
        }

        JFrame frame = new JFrame("Test Embedding Progress Bar");

        JMenuBar menuBar = new JMenuBar();
        frame.setJMenuBar(menuBar);
        menuBar.add(new JMenu("File"));
        menuBar.add(new JMenu("Edit"));
        menuBar.add(new JMenu("View"));
        menuBar.add(new JMenu("Options"));
        menuBar.add(new JMenu("Help"));
        menuBar.add(new JMenu("About"));

        menuBar.add(Box.createHorizontalGlue());
//        menuBar.add(new JButton("Test button"));
        JProgressBar progressBar = new JProgressBar();
        progressBar.setString("Test Progress Bar");
        progressBar.setStringPainted(true);
        menuBar.add(progressBar);

        frame.pack();
        frame.setExtendedState(Frame.MAXIMIZED_BOTH);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setLocationRelativeTo(null);
        frame.setVisible(true);
    }
}

```

**Comment by DevCharly** [comment]
To reduce the width of the progress bar, you have to set its maximum size. Otherwise `BoxLayout` stretches it similar to the glue.

E.g.:
~~~java
progressBar.setMaximumSize( progressBar.getPreferredSize() );
~~~

Anyway it should not overlap the title. I'll have a look...

**Comment by grimlock81** [comment]
Setting the max size to the preferred size does help, but only if I turn on  `progressBar.putClientProperty("JProgressBar.largeHeight", true);` otherwise it's all squashed.

Do you know if there is a way of allocating a certain amount of the remaining space to the progress bar? For example, if I wanted the progress bar to take one-quarter of the space between the last Menu title and the 'Minimise Window' button. This sounds like a job for a LayoutManager but not sure if changing it on the JMenuBar (the BoxLayout as you were saying) would interfere with its main job of rendering JMenus.

Basically in my case, setting the max size to the preferred size isn't ideal as the String painted on the progress bar will be changing during the app's runtime.

**Comment by DevCharly** [comment]
The overlapping is now fixed in main branch.

Regarding your question:
Using preferred size was only an example.
It is only important to set the **maximum width**. Otherwise BoxLayout stretches the progress bar.

To keep the maximum height, overriding `getMaximumSize()` and modify only maximized width works fine. E.g:

~~~java
JProgressBar progressBar = new JProgressBar() {
    @Override
    public Dimension getMaximumSize() {
        Dimension size = super.getMaximumSize();
        return new Dimension( UIScale.scale( 100 ), size.height );
    }
};
~~~

To have a variable progress bar width, you can use e.g. following in `getMaximumSize()`:
~~~java
return new Dimension( Short.MAX_VALUE / 4, size.height );
~~~

The horizontal glue component has a maximum width of `Short.MAX_VALUE`.
`BoxLayout` seems to distribute space depending on maximum component width.
The larger the maximum width is, the more space it gets.
Using `Short.MAX_VALUE / 4` (or similar) for the progress bar makes it about 1/4 the width of the glue.
This behavior probably also depends on preferred width, because on smaller window widths the progress bar becomes larger than the glue.
